src/workloads/query/TelemetryQueryShapes.yml

```diff
@@ -0,0 +1,129 @@
+
+KeyWords:
+- telemetry
+- query
+- shape
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 400
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  Repeat: &Repeat 1000
+  Threads: &Threads 32
+  MaxPhases: &MaxPhases 6
+  nop: &Nop {Nop: true}
+  dbname: &db test
+
+ActorTemplates:
+- TemplateName: ComputeFindQuery
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "ComputeFindQuery"}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: 1}}
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Database: *Database
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: {a: 1, b: 2, x0: {$exists: false}}}}
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a1: &integer {^RandomInt: {min: -100, max: 100}}
+          a2: *integer
+        Indexes:
+        - keys: {a1: 1, a2: 1}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+# 32 threads with distinct query shapes
+- ActorFromTemplate:
+    TemplateName: ComputeFindQuery
+    TemplateParameters:
+      Name: "AlternatingShapes"
+      Threads: 32
+      Filter:
+        {$and:
+           [{^Object: {
+             withNEntries: 3,
+             havingKeys: {^Join: {array: ["a", {^Choose: {from: ["1", "2", {^RandomString: {length: 10}}], deterministic: true}}], sep: ""}},
+             andValues: {^Choose: {from: [1, 1, {$exists: false}], deterministic: true}},
+             duplicatedKeys: skip
+           }}]
+        }
+      OnlyActiveInPhase: 3
+
+# 32 threads with the same query shape
+- ActorFromTemplate:
+    TemplateName: ComputeFindQuery
+    TemplateParameters:
+      Name: "OneShape"
+      Threads: 32
+      Filter: 
+        {$and:
+           [{a1: 1, a2: 1, x0: {$exists: false}}]
+        }
+      OnlyActiveInPhase: 4
```

Comment:
> @cswanson310 I changed the workload to have 2 actors, one which oscillates between query shapes on 32 threads, and one that runs the same shape on 32 threads. It actually seems like oscillating shapes had worse performance, which is different from my initial analysis with the past workload. I was thinking maybe the extra time taken could be due to the building of the object / creating the random field name, but I ran the workload with the one shape query going through the same process and am still getting the same results. Here is the link to the new sheet with the results: https://docs.google.com/spreadsheets/d/1_cLQ5sZ3A-4ag05b46n1GZAKSTTzm_YzUqeBgSXFCOg/edit#gid=0

Another theory: this "OneShape" actor is going second - that might be an advantage because the first actor will pull the relevant data into the cache, making this one go slightly faster. Could you swap the order as an experiment? Or even just duplicate the "AlternatingShapes" in phase 5 as well as phase 3 to see if they have different performance?

Reply:
If that turns out to be true - the next experiment might be to see if adding another quiesce stage between the two could help. I'm not really sure it is expected to help with flushing caches, it may just be about flushing to disk for insert/writing stability.